### PR TITLE
MAINT: cleanup the lasts of datetime.utcnow()

### PR DESCRIPTION
**Pull request by bsipocz**
### Description

I'm chasing some loose datetime.datetime.utcnow() deprecation warning in some test suites, and one of these was seemingly coming from sqlalchemy. It wasn't, but nevertheless these minor cleanup changes may still be found useful.

### Checklist

This pull request is:

- [x] A documentation / typographical / small typing error fix
	- Good to go, no issue or tests are needed

**Have a nice day!**

### CODE REVIEW

**Review by CaselIT — CHANGES_REQUESTED** [review]
Thanks for noticing, left a comment

File: examples/extending_query/temporal_range.py [hunk 1]
```diff
@@ -23,7 +23,7 @@ class HasTemporal:
     """Mixin that identifies a class as having a timestamp column"""
 
     timestamp = Column(
-        DateTime, default=datetime.datetime.utcnow, nullable=False
+        DateTime, default=datetime.datetime.now(datetime.UTC), nullable=False
```
[the remark this answers is not in the document]
Comment (bsipocz) [reply examples/extending_query/temporal_range.py]: Indeed, thanks for spotting. 

I also noticed that this is not python <3.11 compatible, that has hopefully been fixed with the new commits, but this being my first contribution here CI has to be authorized by a maintainer to run.

File: examples/extending_query/temporal_range.py [hunk 2]
```diff
@@ -22,9 +23,8 @@
 class HasTemporal:
     """Mixin that identifies a class as having a timestamp column"""
 
-    timestamp = Column(
-        DateTime, default=datetime.datetime.utcnow, nullable=False
-    )
+    utc = partial(datetime.datetime.now, datetime.timezone.utc)
```
Comment (CaselIT) [inline examples/extending_query/temporal_range.py:26]: I would avoid assigning this function inside the class

File: tox.ini [hunk 3]
```diff
@@ -101,6 +101,7 @@ setenv=
     PYTEST_COLOR={tty:--color=yes}
     MYPY_COLOR={tty:--color-output}
 
+    PYTEST_ARGS=
```
Comment (CaselIT) [inline tox.ini:104]: Can you please remove this? Thanks

Comment (bsipocz) [reply tox.ini:104]: I'll remove the commit for this, but I added this as I was running into issues running tox locally for python <3.12. Anyway, I think #11838 can be done now.

File: examples/extending_query/temporal_range.py (hunk 2 above)
Comment (bsipocz) [reply examples/extending_query/temporal_range.py]: Done

**Comment by CaselIT** [comment]
looks ok now! thanks

**Review by CaselIT — APPROVED** [review]

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work on behalf of **CaselIT** to try to get revision 9bee8af8d1082c3cde5f64c78f1e565ef4ab14cd of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by sqla-tester** [comment]
New Gerrit review created for change 9bee8af8d1082c3cde5f64c78f1e565ef4ab14cd: https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/5501

**Comment by sqla-tester** [comment]
Gerrit review https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/5501 has been **merged**. Congratulations! :)

**Comment by sqla-tester** [comment]
Gerrit review https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/5503 has been **merged**. Congratulations! :)

**Pull request closed without merging**